### DISCUSSION TITLE
In the NPT simulation, it expands to gas above the melting points

### DISCUSSION THREAD

**Discussion by pobo95** [ACEsuit/mace#317] [Q&A] [answered]
When I tested NPT simulation above the melting points

it expands to gas but it didn't when I used QUIP ML potential

![image](https://github.com/ACEsuit/mace/assets/83407140/903a3ff2-4827-4f66-b3b6-e4aaa17ebfd6)

here is my trianing parameters

E0s = average
num_interactions = 2
correlation = 2
max_L = 1
r_max = 5.0
hidden_irreps = 64x0e + 64x1o

full command
`mace_run_train     --name="mace"     --train_file="train.xyz"     --valid_fraction=0.05     --test_file="validate.xyz"     --E0s="average"     --model="MACE"     --num_interactions=2     --hidden_irreps='64x0e + 64x1o'     --max_L=1     --correlation=2     --r_max=5.0     --forces_weight=1000     --energy_weight=10     --batch_size=5     --valid_batch_size=2     --max_num_epochs=5000   --compute_stress=”true”  --start_swa=450     --scheduler_patience=5     --patience=15     --eval_interval=3     --ema     --swa     --swa_forces_weight=10        --default_dtype="float64"    --device=cuda     --seed=123     --restart_latest     --save_cpu --error_table=PerAtomRMSEstressvirials`

It shows 
RMSE energy per atom : 1.1 meV/atom 
RMSE force per atom : 81.8 meV/A
RMSE stress per atom : 0.01 meV/A^3
for test file

I attach my traing and validate files too

[train_validate.zip](https://github.com/ACEsuit/mace/files/14167040/train_validate.zip)

Do you have any clues to slove it?

Thanks

**Comment by gabor1** [comment, marked as answer]
yes. what's happenning here is that your isolated atom energies are not correct. you are using the "average" method for getting E0s. if you calculate isolated atom energies, and tell MACE to use those instead, you will not see the evaporation. You are saying that "QUIP ML", by which I guess you mean SOAP/GAP, didn't evaorate - perhaps you used calculated isolated atom energies for E0s there ?

**Reply by pobo95** [reply]
Thank you for your quick answer.
That's correct I calculated isolated atom energys for E0s when I used SOAP/GAP. I will try trianing again with exact E0s.

Does isolated atom energy matter when calculating pressure?

**Comment by gabor1** [comment]
the model is completely different when you include real E0s, so yes, your stress predictions will change.

**Reply by pobo95** [reply]
If I rememeber correctly, as far as I know, E0s are using interatomic potential to fit binding energy.
Isn't this included in the massage passing process?
What I'm saying is, the massage passing process already contains information about binding energy?

**Reply by gabor1** [reply]
"E0s are using interatomic potential to fit binding energy"  - I can't parse this. the model fits binding energy ONLY if you explicitly include E0s, rather than using "average". in order to have stable NPT you need to fit binding energies.

**Reply by pobo95** [reply]
Thank you for your answer. I belived massage passing process can describe binding energies.
I leaned a lot thank you again!

**Reply by bernstei** [reply]
To put it slightly more precisely, for a potential have to correct atomization energy limit you need to fit binding energies, which means that your E0s need to be physically meaningful isolated atom energies (on the same scale as the rest of the reference data), rather than data averages (MACE `run_train --E0s average`). It's possible to have stable NPT (at least for a limited range of pressures) without this, but it's much harder to achieve.

**Reply by pobo95** [reply]
Thank you so much. It helped me a lot. I will retrain my potentials with E0s. I think I've solved the problem thank you :)

**Reply by gabor1** [reply]
> I belived massage passing process can describe binding energies.

it can, but only if you give E0s explicitly. it cannot guess the correct E0s, there is absolutely nothing in your training data that can determine what the isolated atom energies are (unless you have the isolated atoms in the training data set and are set up to be treated in a special way, but we don't especially look for it currently)

**Reply by bernstei** [reply]
If you include configs with isolated atoms and set `config_type=IsolatedAtom` (or something like that), and you don't pass any `--E0s` flag, I believe it uses those configs.

**Reply by pobo95** [reply]
I got it, Thank you maybe I can test include isolated atoms in my training data :)

**Reply by pobo95** [reply]
> If you include configs with isolated atoms and set config_type=IsolatedAtom (or something like that), and you don't pass any --E0s flag, I believe it uses those configs.

@bernstei I just check it in the utils.py file thank you for remind. 
BTW when I use `--E0s={32:-0.02859140,51:-0.03305661,52:-0.03357790}` flag it only read the last one (52:-0.03357790) in mace = 0.3.4 v, python = 3.9.7 v. Maybe somthing wrong in my system.
